File: backend/src/main/java/com/example/backend/domain/voter/dto/VoterDTO.java
package com.example.backend.domain.voter.dto;

import com.example.backend.domain.voter.entity.Voter;
import com.example.backend.domain.vote.entity.Vote;
import com.example.backend.domain.member.entity.Member;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class VoterDTO {
    private Long memberId;  // 투표자 ID
    private Long voteId;  // 투표 ID

    // Entity -> DTO
    public static VoterDTO from(Voter voter) {
        VoterDTO voterDto = new VoterDTO();
        voterDto.memberId = voter.getId().getMemberId();
        voterDto.voteId = voter.getId().getVoteId();
        return voterDto;
    }

    public Voter toEntity(Member member, Vote vote) {
        return Voter.builder()
                .id(new Voter.VoterId(memberId, voteId))
                .member(member)
                .vote(vote)
                .build();
    }
}

File: backend/src/main/java/com/example/backend/domain/voter/dto/VoterDTO.kt
package com.example.backend.domain.voter.dto

import com.example.backend.domain.voter.entity.Voter
import com.example.backend.domain.vote.entity.Vote
import com.example.backend.domain.member.entity.Member

data class VoterDTO(
    val memberId: Long,
    val voteId: Long
) {
    companion object {
        fun from(voter: Voter): VoterDTO {
            return VoterDTO(
                memberId = voter.id.memberId,
                voteId = voter.id.voteId
            )
        }
    }

    fun toEntity(member: Member, vote: Vote): Voter {
        return Voter(
            id = Voter.VoterId(memberId, voteId),
            member = member,
            vote = vote
        )
    }
}
